### PR TITLE
Update stage number when stages have been removed (BL-2464)

### PR DESCRIPTION
**Pull request by StephenMcConnel**
Also clear the letter display if the stage count goes to zero.

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts [hunk 1]
```diff
@@ -370,7 +373,11 @@ class ReaderToolsModel {
    */
   updateLetterList(): void {
     var stages = this.synphony.getStages();
-    if (stages.length === 0) return;
+    if (stages.length === 0) {
+      // In cause the user deletes all stages, and something had been displayed before.
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts]: cause -> case?

File: src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts [hunk 2]
```diff
@@ -123,6 +123,9 @@ class ReaderToolsModel {
       ReaderToolsModel.updateElementContent("stageNumber", "0");
       return;
     }
+    if (this.stageNumber > stages.length) {
+       this.stageNumber = stages.length;
+    }
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts]: Is it possible for them to delete a stage other than the last? Seems if they delete the last stage, the others would not need adjusting, and if they can delete an earlier one, all the following ones would need adjusting, not just the one that is out of range.

Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts]: Things adjust without weirdness or crashing if the selected stage is not greater than the number of stages.  I suppose in some sense if stage 3 was selected and the user deleted stages 1 and 2, the selected stage should always change to stage 1 even if we have 17 stages still in existence.  But we don't have that information available here, and I don't have a good enough understanding of the js code to achieve that happy state.

File: src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts (hunk 1 above)
Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts]: oops.

**Pull request merged**